docs/frontends/queues.md

```diff
@@ -123,10 +123,28 @@ Internally, our PIFO tree is implemented by leveraging the PIFO frontend.
 The PIFO frontend seeks to orchestrate two queues, which in the simple case will just be two FIFOs.
 However, it is easy to generalize those two queues: instead of being FIFOs, they can be PIFOs or even PIFO trees.
 
+## Minimum Binary Heap
+
+A binary heap is another tree-shaped data structure, where each node has at most two children.
+However, unlike the previous queues, a heap exposes an extended interface: 
+in addition to our previous input ports and reference registers, a heap has an additional input `rank`.
+The `push` operation now accepts both a `value` and its associated `rank`. 
+Consequently, a heap maintains an ordering of elements by `rank`, with the `pop` and `peek` operations removing and reading an element with minimal rank.
+
+To maintain this ordering efficiently, a heap stores `(rank, value)` pairs in each node and takes special care to uphold the following invariant:
+> **Min-Heap Property**: for any given node `C`, if `P` is a parent node of `C`, then the rank of `P` is less than or equal to the rank of `C`.
+
+This allows `peek` to be constant time and `push` and `pop` to be logarithmic in the size of the heap.
+
+Our frontend allows for the creation minimum binary heaps of any height, that operate on values and ranks of any width, in Calyx. 
+  - See [here][binheap.py] for an example heap of height four, operating on 32-bit values and 64-bit ranks.
+  - See [here][stable_binheap.py] for a thin layer over our heap that breaks rank ties via insertion order. 
```

Comment:
I think this deserves a little paragraph of its own. Please explain that the standard binary heap has unclear semantics in case of rank-based ties, but it is often worthwhile to break such ties in FIFO order. Then explain briefly the implementation trick. Also, right now the actual words "stable binheap" won't appear on the webpage, so a user won't be able to search for that. Consider just saying that name and linking off that, not off "here" 🙃

Reply:
The most recent commit attempts to include the details requested here; let me know what you think!

Reply:
I took another crack at it. See what you think, and then feel free to merge!

Reply:
I like it; I think your changes add a lot of clarity